crates/nu-command/src/filesystem/rm.rs

```diff
@@ -377,11 +362,7 @@ fn rm(
                     );
 
                     let result;
-                    #[cfg(all(
-                        feature = "trash-support",
-                        not(target_os = "android"),
-                        not(target_os = "ios")
-                    ))]
+                    #[cfg(trash)]
```

Comment:
Could we simplify those two blocks that only differ in one `else if` by making use of a `cfg!` expression macro and/or a function with the conditional compilation?

Reply:
Good idea, I didn't look too closely at the logic here between these two blocks, but I see now there was a ton of repetition.

I consolidated them into one block, and removed the `build.rs` in favor of this and a constant. Originally I did not like the idea of pushing this into runtime, but the more I thought about it, I realized that if it's a const, then stuff like `if false { ... }` probably gets compiled out by the optimizer anyway, so it's probably not having any effect on the final resulting code.